src/mwoffliner.lib.ts

```diff
@@ -290,7 +291,7 @@ async function execute(argv: any) {
   let articleListLines: string[]
   if (articleList) {
     try {
-      articleListLines = await readFileOrUrlByLine(articleList)
+      articleListLines = await readFileOrUrlByLine(articleList, downloader.streamRequestOptions, tmpDirectory)
```

Comment:
Please don't add additional parameter only for technical reason:
* There is a global tmp directory defined, use it
* considering that the downloader is always the same, no reason to pass it as argument.

Reply:
I moved the `readFileOrUrlByLine()` function into `misc.ts` because, from the old place, it cannot be tested simply.
The temporary directory is not globally defined, it is defined in `mwoffliner.lib.ts`, and after executing the script all temp files are removed. That's why I added it as a parameter.
Also, tests use another tmp directory.

Same situation with the downloader. The Downloader object creates inside `execute()` function in the file `mwoffliner.lib.ts`. `streamRequestOptions` method depends on downloader params which are passed to constructor.

If I remove both parameters then I need to move the `readFileOrUrlByLine()` function back inside the `execute()` function in the file `mwoffliner.lib.ts`. After this, I can't test it because it will be function inside another function.

Reply:
Do the revamping work that all of this fits propely…. If you have to add useless parameters like this, that just mean you have a deeper code organisation problem. A problem you will have to fix anyway at some point, so just do it early.